src/web/pages/portlists/ListPage.jsx

```diff
@@ -145,23 +145,24 @@ const PortListsPage = () => {
   );
 
   useEffect(() => {
-    // load initial data
-    if (
+    /*
+     * Check if the filters are changed in a way that matters for fetching:
+     * - Either the the previous filter wasn't defined yet
+     * - or the terms are different
+     * - or the normal equality check fails (for cases where both filters have IDs).
+     */
+
+    const shouldFetch =
       isDefined(filter) &&
       !isLoadingFilter &&
-      !filter.equals(previousFilter)
-    ) {
+      (!isDefined(previousFilter) ||
+        filter.toFilterString() !== previousFilter?.toFilterString() ||
+        !filter.equals(previousFilter));
```

Comment:
I am a bit irritated about this change. It shouldn't behave differently then before.
Both `!isDefined(previousFilter)` and `filter.toFilterString() !== previousFilter?.toFilterString()` should be covered by `!filter.equals(previousFilter)` already.